plugins/services/src/js/structs/ServiceTree.js

```diff
@@ -166,6 +167,32 @@ module.exports = class ServiceTree extends Tree {
     }, 0);
   }
 
+  getStatusIconCounts() {
+    const summaryCounts = {
+      status: {},
+      total: 0
+    };
+
+    return this.reduceItems(function(counts, item) {
+      if (item instanceof Service) {
+        const itemStatus = item.getServiceStatus();
+        if (itemStatus === null) {
+          return counts;
+        }
+        counts.total++;
+
+        const iconName = itemStatus.icon.name;
```

Comment:
as @GeorgiSTodorov found out: `itemStatus.icon` can be null (e.g. for Services with the status "N/A").
we should add a safeguard here.